app/models/order.rb

```diff
@@ -0,0 +1,8 @@
+class Order < ActiveRecord::Base
+  belongs_to :user
+  has_many :line_items
+
+  def purchase_line_items
+    LineItem.where(cart: user.cart).update_all(cart_id: nil, order_id: id)
```

Comment:
You can simplify this to `user.cart.line_items.update_all ...` without losing efficiency.